app/styles/ui/history/_commit-list.scss

```
@@ -52,6 +52,16 @@
     }
   }
 
+  &.has-highlighted-commits {
+    .list-item {
+      transition: opacity 0.6s;
+
+      &:not(.highlighted) {
+        opacity: 30%;
+      }
+    }
+  }
+
```

I personally think we should also make a faster transition when nothing is highlighted:

https://user-images.githubusercontent.com/1083228/176643427-0e46afd4-dbaf-4191-b3d1-7da258dc9af8.mov

These are the changes I applied:
```suggestion
  &.has-highlighted-commits {
    .list-item {
      transition: opacity 0.5s;

      &:not(.highlighted) {
        opacity: 30%;
      }
    }
  }

  &:not(.has-highlighted-commits) {
    .list-item {
      transition: opacity 0.25s;
    }
  }

```